internal/reconcilers/run_profile.go

```diff
@@ -92,88 +88,30 @@ func (r *Reconciler) publishProfileInitEvents(
 	ctx context.Context,
 	projectID uuid.UUID,
 ) error {
-	dbrepos, err := r.store.ListRegisteredRepositoriesByProjectIDAndProvider(ctx,
-		db.ListRegisteredRepositoriesByProjectIDAndProviderParams{
-			Provider:  sql.NullString{Valid: false},
-			ProjectID: projectID,
-		})
+	ents, err := r.store.GetEntitiesByProjectHierarchy(ctx, []uuid.UUID{projectID})
```

Comment:
2 questions:
1) Is it OK to list all entities? We used to only list repos and artifacts, this would also include pulls. I don't see why we shouldn't evaluate pulls that are already tracked in our database, but it is a change in behaviour
2) the old code doesn't seem to take hierarchy into account, in other words it would only trigger evaluation of entities in that project only, but not subprojects. Is that intended?

Reply:
1. Makes sense and I think it's ok.
2. That was an issue tracked somewhere... thanks for fixing it!

Reply:
about 2 - I haven't actually searched the child projects to get the hierarchy, I just prepped the code to accept a hierarchy but pass a single-item slice. I'll push another patch to this PR for easier review.

Reply:
I think not doing the child hierarchy in this PR is the right thing to do. It is a TODO item and we should address it, but let's keep this PR small.